src/query_view.js

```diff
@@ -56,6 +60,47 @@ class QueryView extends SelectListView {
       });
     }
   }
+
+  getIcon_(slug) {
+    this.lazyLoadIcons_();
+
+    // find most specific available icon
+    // first consider the full slug
+    var indices = [slug.length];
+    // then consider the substring without the minor version
+    if (slug.indexOf('.') != -1) {
+      indices.push(slug.indexOf('.'));
+    }
+    // then consider the substring without the version
+    if (slug.indexOf('~') != -1) {
+      indices.push(slug.indexOf('~'));
+    }
+    for (var i = 0; i < indices.length; ++i) {
+      var name = slug.substring(0, indices[i]);
+      if (Icons.indexOf(name) != -1) {
+        return name;
+      }
+    }
+    // fallback to the full slug (even if it doesn't exist)
+    // optionally this could show a default icon?
+    return slug;
+  }
+
+  lazyLoadIcons_() {
```

Comment:
Agreed. Maybe you can rename the function / variable to indicate that only file names are being cached and not the icons themselves. Skimming through the code, it looks like there's an entire icon cache and not just a file name cache.